distributed/scheduler.py

```diff
@@ -3158,6 +3160,12 @@ def coerce_address(self, addr, resolve=True):
 
         return addr
 
+    def clear_ip_cache(self, addr):
+        if isinstance(addr, tuple):
+            addr = unparse_host_port(*addr)
+            clear_address_cache(addr)
```

Comment:
What should happen otherwise?  Should we raise a NotImplementedError?  The silent behavior here is probably dangerous for future developers.

Reply:
We should largely mirror the logic in coerce_address, as that is how addresses get fed into caches. (But exclude the check for aliases, as (1) they are pre-resolved, and (2) this would miss the underlying cache that would still be there after the alias gets deleted)